mono/mini/mini.c

```diff
@@ -2953,16 +2940,16 @@ mono_insert_safepoints (MonoCompile *cfg)
 	 * we can skip the GC safepoint on method entry. */
 	gboolean requires_safepoint = cfg->has_calls;
 
+	if (requires_safepoint)
+		insert_safepoint (cfg, cfg->bb_entry);
```

Comment:
why was this moved? This should be after the loop (it potentially changes `requires_safepoint`).